bitcoin/contrib/test.sh

```diff
@@ -53,20 +60,25 @@ then
     fi
 fi
 
-echo "********* Testing std *************"
-# Test without any features other than std first
-cargo test --locked --verbose --no-default-features --features="std"
-
-echo "********* Testing default *************"
-# Then test with the default features
-cargo test --verbose
-
-# Test each feature
-for feature in ${FEATURES}
-do
-    echo "********* Testing $feature *************"
-    cargo test --locked --verbose --features="$feature"
-done
+if [ "$DO_FEATURE_MATRIX" = true ]; then
+    cargo build --locked --no-default-features
+    cargo test --locked --no-default-features
+
+    # All features
+    cargo build --locked --no-default-features --features="$FEATURES"
+    cargo test --locked --no-default-features --features="$FEATURES"
+    # Single features
+    for feature in ${FEATURES}
+    do
+        cargo build --locked --no-default-features --features="$feature"
+        cargo test --locked --no-default-features --features="$feature"
+		# All combos of two features
+		for featuretwo in ${FEATURES}; do
+			cargo build --locked --no-default-features --features="$feature $featuretwo"
+			cargo test --locked --no-default-features --features="$feature $featuretwo"
+		done
```

Comment:
Yeah, this is probably a good idea. While formally we would have to test all combinations (2^feature_count) that's too much and combinations of two make sense - breakage caused by combination of three or more is unlikely.

Reply:
Yeah, testing every pair is great. IIRC my old code only tested single features (and all/none).